lib/utils/templatize.html

```diff
@@ -471,14 +475,8 @@
        * always be set to the instance (regardless of whether they would normally
        * be forwarded from the host).
        *
-       * Note that the class returned from `templatize` is generated only once
-       * for a given `<template>` using `options` from the first call for that
-       * template, and the cached class is returned for all subsequent calls to
-       * `templatize` for that template.  As such, `options` callbacks should not
-       * close over owner-specific properties since only the first `options` is
-       * used; rather, callbacks are called bound to the `owner`, and so context
-       * needed from the callbacks (such as references to `instances` stamped)
-       * should be stored on the `owner` such that they can be retrieved via `this`.
```

Comment:
In addition to the limitation you added here, we should clarify the point here, not delete it.  It is nuanced:

Consider the following:

```html
<dom-repeat items="[[items]]" id="outerRepeat">
  <template id="outerTemplate">
    <dom-repeat items="[[item.items]]" id="innerRepeat">
      <template id="innerTemplate">
        <div>[[item.text]]</div>
      </template>
    </dom-repeat>
  </template>
</dom-repeat>
```
The `innerRepeat` will call `templatize` to retrieve a constructor to generate instances of `innerTemplate` once for each outer item (and then stamp instances from the constructor returned for each inner item).  Technically it is a new instance of a `<template id="innerTemplate">` that is being templatized for each outer item, and so does not violate the rule you replaced it with ("can only be run once for a given `<template>`").

However there's an additional limitation that the note you deleted was trying to capture.  Templatizer caches the class generated from the first call to `templatize` for `innerTemplate`, and returns that generated class for every subsequent call to `templatize` on `innerTemplate`.  Without this caching, for every instance of `outerRepeat`, the `innerRepeat` would re-templatize the inner template, which is wasteful since we can track that the template is identical.  However, this caching means that the `options` passed into `templatize` are only respected for the first call of `templatize` for a given _nested template_.

Perhaps emphasizing this note applies to _*nested*_ templates, while adding additional note you provided, would be best.

Reply:
Thanks for the clarification! I'll update this PR with a revised explanation. There's no doubt that this should be documented because there's a lot going on.

I'll update this PR tomorrow. (It's midnight now here in Germany)